pep-0572.rst

```diff
@@ -126,11 +130,14 @@ that value::
 
     # Handle a matched regex
     if (match := pattern.search(data)) is not None:
-        ...
+        # Do something with match
 
-    # A more explicit alternative to the 2-arg form of iter() invocation
-    while (value := read_next_item()) is not None:
-        ...
+    # A loop that can't be trivially rewritten using 2-arg iter()
+    while (value := read_next_item(file)) is not None:
+        # Do something with value
```

Comment:
Ick - this example seems wholly contrived.  How about keeping the original `# A more explicit ...` comment but changing the example to match it?

    Before:
        for line in iter(somefile.readline, ""):
            process(line)
    After:
        while line := somefile.readline():
            process(line)

Maybe that just especially appeals to me because I can never remember what 2-arg `iter()` does ☹️

Reply:
The problem with file based examples is that they typically collapse down to `for line in somefile: ...` in real code. It really is only in cases where native iteration support isn't available for some reason that you need to resort to the loop-and-a-half pattern instead.

Reply:
Another real-world example involving files is:

‘’’
while chunk := file.read(8192):
     process(chunk)
‘’’

Reply:
@giampaolo, I think that would be an excellent example to replace the one here!  The clumsy 2-arg iter way of spelling that is

    for chunk in iter((lambda: somefile.read(8192)), ""):

Reply:
Adding this now.